src/sessions/live-monitoring-session-handler.ts

```diff
@@ -0,0 +1,63 @@
+import {
+  IPendingSession,
+  IExtendedMediaSession,
+  IAcceptSessionRequest,
+  IUpdateOutgoingMedia,
+} from '../types/interfaces';
+import BaseSessionHandler from './base-session-handler';
+import { SessionTypes, SdkErrorTypes } from '../types/enums';
+import {createAndEmitSdkError, isLiveScreenMonitorJid} from '../utils';
+import {Constants} from "stanza";
+
+export class LiveMonitoringSessionHandler extends BaseSessionHandler {
+  sessionType = SessionTypes.liveScreenMonitoring;
+
+  shouldHandleSessionByJid(jid: string): boolean {
+    return isLiveScreenMonitorJid(jid);
+  }
+
+  handleConversationUpdate(): void {
+    /* no-op */
+    return;
+  }
+
+  async handlePropose (pendingSession: IPendingSession): Promise<void> {
+    if (this.sdk._config.autoAcceptPendingLiveScreenMonitoringRequests) {
+      return this.proceedWithSession(pendingSession);
+    }
+
+    // if not auto accepting sessions, emit this event for the consumer to accept
+    await super.handlePropose(pendingSession);
+  }
+
+  async acceptSession(session: IExtendedMediaSession, params: IAcceptSessionRequest): Promise<any> {
+    if (!params.mediaStream) {
+      throw createAndEmitSdkError.call(this.sdk, SdkErrorTypes.session, `Cannot accept live screen monitoring session without providing a media stream`, { conversationId: session.conversationId, sessionType: this.sessionType });
+    }
+
+    // Set outbound stream to primary video media stream
+    session._outboundStream = params.mediaStream;
+
+    let addMediaPromise: Promise<any> = Promise.resolve();
+    params.mediaStream.getTracks().forEach((track) => {
+      addMediaPromise = addMediaPromise.then(() => {
+        this.sdk.logger.info('Adding screen track to session', { trackId: track.id, label: track.label, conversationId: session.conversationId, sessionType: this.sessionType });
```

Comment:
It's probably worth prefixing this log with 'Live monitorins - adding screen track to session' for debugging purposes.